sshuttle/client.py

```diff
@@ -643,9 +651,24 @@ def matches_acl(dstip, dstport, store_to_check):
 
     return False
 
+def ipInNetwork(ip, cidrBlock):
+    try:
+        network = ipaddress.ip_network(cidrBlock, False)
+        reqIp = ipaddress.ip_address(ip)
+        if reqIp in network:
+            return True
+    except:
+        log('Failed to parse CIDR block %s' % cidrBlock)
+    return False
+
 def tcp_connection_is_allowed_conditional(dstip, dstport, srcip, check_acl, check_sources):
 
     if check_sources:
+        # allow any IP from the two Microservice subnets
+        if ipInNetwork(srcip, MS1_CIDR) or ipInNetwork(srcip, MS2_CIDR):
+            debug3("TCP-Source request is from the microservice subnet\n")
```

Comment:
Since you have the information available it would be worth including the source IP in the log.